### DISCUSSION TITLE
Wrong execution date in SLA alerts

### DISCUSSION THREAD

**Discussion by hliu-VoleonGroup** [apache/airflow#39840] [Q&A]
### Apache Airflow version

main (development)

### What happened

- The execution date in a triggered SLA alert is one schedule interval ahead of the expected one. 

- In other words, the execution date of the SLA record inserted into the `SlaMiss` table is incorrect.

- For example, if a daily task with an execution date `2023-10-23 21:00:00` triggers an SLA alert, the execution date in that alert will be `2023-10-24 21:00:00` rather than `2023-10-23 21:00:00`.

### What you think should happen instead

The execution date of the dagrun shown in an SLA alert should be the same as the one that triggered the alert.

### How to reproduce

1. Create a simple DAG scheduled every 5 minutes

```
import datetime
from airflow.models import DAG
from airflow.operators.bash import BashOperator

DEFAULT_ARGS = {
    "owner": "test",
    "email": ["test@email.com"]
}

dag = DAG(
    dag_id="test_dag",
    schedule_interval="*/5 * * * *",
    start_date=datetime.datetime(2023, 10, 20),
    catchup=False
)

with dag:
  task = BashOperator(
      task_id="task",
      bash_command="sleep 120; dummy command",
      sla=datetime.timedelta(minutes=1))
```

2. Enable the DAG and mark its first dagrun with execution date T as success.
3. Its second dagrun with execution date T+5 should fail and trigger an SLA alert at T + 11 (T + 5 + 5 + 1).
     However, the execution date of the task in this alert will be T + 10 instead of T + 5, which is not correct.
4. Also, the SLA record in the `SlaMiss` table has a wrong execution date as well.

### Operating System

CentOS Linux 7

### Versions of Apache Airflow Providers

apache-airflow-providers-common-sql==1.8.0
apache-airflow-providers-ftp==3.6.0
apache-airflow-providers-http==4.6.0
apache-airflow-providers-imap==3.4.0
apache-airflow-providers-sqlite==3.5.0

### Deployment

Other Docker-based deployment

### Anything else

I have submitted a PR to apache-airflow repository for fixing this issue.
https://github.com/apache/airflow/pull/35389
It would be great if someone could take a look.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by pateash** [comment]
hi @hliu-VoleonGroup 
this logicalDate represents(logical date of next dagrun) or  the Data interval end time ( when SLA was fired ).

<img width="719" alt="image" src="https://github.com/apache/airflow/assets/16856802/577dcda3-baf3-425a-971d-8e396b41c200">

Although I agree we should have clear naming.
Additionally we can also add one more column which show schedule date as well.
cc. @potiuk @ephraimbuddy  @uranusjr

**Comment by hliu-VoleonGroup** [comment]
@pateash Thanks for the reply.
To my understanding, `logicalDate` is a new concept in Airflow 2 and it is equivalent to `executionDate`. Please correct me if I am wrong.

Besides, I think it's a behavior change from Airflow 1 to Airflow 2. In Airflow 1, the date in SLA alert was correct which is the executionData of the task. However, in Airflow 2, it becomes one interval ahead of the task that triggers the SLA.
